frontend/packages/integration-tests-cypress/tests/i18n/pseudolocalization.spec.ts

```diff
@@ -14,38 +20,134 @@ describe('Localization', () => {
+  const createExampleResource = (url: string) => {
+    listPage.clickCreateYAMLbutton();
+    cy.byTestID('resource-sidebar').should('exist');
+    yamlEditor.isLoaded();
+    yamlEditor.clickSaveCreateButton();
+    cy.get(errorMessage).should('not.exist');
+    cy.visit(url);
+  };
 
-  it('pseudolocalizes activity card', () => {
-    cy.log('test activity card components');
-    cy.visit('/dashboards?pseudolocalization=true&lng=en');
-    cy.byTestID('activity').isPseudoLocalized();
-    cy.byTestID('activity-recent-title').isPseudoLocalized();
-    cy.byTestID('ongoing-title').isPseudoLocalized();
-    cy.byTestID('events-view-all-link').isPseudoLocalized();
-    cy.byTestID('events-pause-button').isPseudoLocalized();
-  });
+  const deleteExampleResource = (kind: string) => {
+    detailsPage.clickPageActionFromDropdown(`Delete ${kind}`);
+    modal.shouldBeOpened();
+    modal.submit();
+    modal.shouldBeClosed();
+  };
 
-  it('pseudolocalizes utilization card', () => {
-    cy.log('test utilization card components');
-    cy.visit('/dashboards?pseudolocalization=true&lng=en');
-    cy.byTestID('utilization-card-item-text').isPseudoLocalized();
-  });
+  const testObjs = OrderedMap<string, LocalizationTestProps>()
+    .set('Masthead', {
+      url: '/settings/cluster',
+      testIDs: ['application-launcher-item'],
+    })
+    .set('Sidebar navigation', {
+      url: '/settings/cluster',
+      testIDs: ['nav'],
+    })
+    .set('Cluster dashboard', {
+      url: '/dashboards',
+      testIDs: [
+        'activity',
+        'activity-recent-title',
+        'ongoing-title',
+        'events-view-all-link',
+        'events-pause-button',
+        'utilization-card-item-text',
+      ],
+    })
+    .set('Events', {
+      url: '/k8s/all-namespaces/events',
+      testIDs: ['timestamp'],
+    })
+    .set('Services list', {
+      url: '/k8s/all-namespaces/services',
+      selectors: [tableColumnHeaders], // all table column headers should be translated
+    })
+    .set('Service details', {
+      url: '/k8s/all-namespaces/services/alertmanager-main',
+      // all section headings, detail item labels, and elements with `data-i18n-translated` attribute should be translated
+      selectors: [sectionHeadings, detailLabels, i18nTranslated],
+      testIDs: ['timestamp'],
+    })
+    .set('Routes list', {
+      url: '/k8s/all-namespaces/routes',
+      selectors: [
+        '.pf-c-dropdown__group-title',
+        '.co-filter-dropdown-item__name',
+        tableColumnHeaders,
+      ],
+    })
+    .set('Route details', {
+      url: '/k8s/all-namespaces/routes/alertmanager-main',
+      selectors: [sectionHeadings, detailLabels, i18nTranslated, '.graph-title'],
+      testIDs: ['timestamp'],
+    })
+    .set('Ingresses list', {
+      url: '/k8s/all-namespaces/ingresses',
+      createResource: true,
+      selectors: [tableColumnHeaders],
+    })
+    .set('Ingress details', {
+      url: '/k8s/all-namespaces/ingresses/example',
```

Comment:
It would be better if each test was independent so it doesn't rely on the previous test creating the resource (or a different test deleting the resource that was created).

We could look at having two groups, one for general pages, and one for resource lists that always create the resource (like the crud tests). Or we can maybe hook this into the existing crud tests.

Reply:
fixed